angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
@thorn0 using your example code, sans `--strictNullChecks`
```ts
angular.injector(['ng']).get<IFilterOrderBy>('orderByFilter')([1, 7, 5, 3] as number[] | null) // number[]

angular.injector(['ng']).get<IFilterOrderBy>('orderByFilter')([1, 7, 5, 3] as number[] | undefined) // number[]
```
which as @smhc noted, is correct.